### ISSUE TITLE
[UX] "Delete and Open" notification button is confusing

### ISSUE THREAD

**Issue by jayschwa** [mozilla-mobile/fenix#6470]
The persistent notification for private tabs has a button labeled "Delete and Open". Deleting a private tab makes sense to me, opening a private tab makes sense to me, but the concept of opening something that has been deleted is confusing at first glance.

* Android device: Google Pixel 2
* Fenix version: 2.3.0

related issue: #5733 

![firefox_preview_private_confusing](https://user-images.githubusercontent.com/475017/68269045-5733ef00-0026-11ea-9266-75412d4098be.png)

**Comment by darkwing** [comment]
An improvement could be to make the label say "Delete open tabs", and upon tapping that text, we delete the tabs, the text goes away, and the user can tap "Open" to open the app.

**Comment by NotWoods** [comment]
cc @AmyYLee

**Comment by AmyYLee** [comment]
> The persistent notification for private tabs has a button labeled "Delete and Open". Deleting a private tab makes sense to me, opening a private tab makes sense to me, but the concept of opening something that has been deleted is confusing at first glance.
> 
>     * Android device: Google Pixel 2
> 
>     * Fenix version: 2.3.0
> 
> 
> ![firefox_preview_private_confusing](https://user-images.githubusercontent.com/475017/68269045-5733ef00-0026-11ea-9266-75412d4098be.png)

@mheubusch - Not sure if you would suggest alternative copy here? on FF Focus we use "erase and open"

The intent of this button is allowing the user to clear their private tabs before opening the app if they don't want their private tabs to be shown upon opening the app.


> An improvement could be to make the label say "Delete open tabs", and upon tapping that text, we delete the tabs, the text goes away, and the user can tap "Open" to open the app.

I think adding an additional tap is not ideal here nor is having a button disappear on tap.

**Comment by Dunexus** [comment]
Last time I used it, I just wanted to closed private tabs without opening Fenix. It was unclear that tapping the notification would delete the tabs so I clicked on "Delete and Open" and then closed the app.

With two buttons below the notification, the focus is on the actions of the buttons.
When other app have a notification with buttons, they usually use tap on the notification to open the app and buttons to do other actions (such as reply or mark as read for messaging apps).

I think it would be less confusing to have "Open" on tap and "Delete" and "Delete and Open" as buttons.

**Comment by jayschwa** [comment]
My personal opinion is that the notification is trying to do too many things. Regular tabs don't have a notification for opening them, so why should private tabs? I think the main purpose of the notification is "Abort!", and having multiple actions muddles that purpose and also adds potential for the user to "misclick" and do an action they didn't want (e.g. open the tabs instead of nuking them). I prefer Chrome's simpler approach of just having the notification nuke private tabs and nothing else.

**Comment by brampitoyo** [comment]
Some background information below.

This dialogue was ported over from Firefox Focus.

Initially, we only had one action:

> 1. Delete private tabs
> Panic, self-destruct button. Equivalent to “Kill app”.

Then we added a secondary action to save time:

> 2. Delete and open
> Equivalent to “Restore app with zero tab”.

Then we added a third action which is not destructive at all:

> 3. Open
> Equivalent to “Restore app with all tabs”.

If I were to redo this feature for Firefox Preview, which is a general-purpose browser, I might consider:

1. Renaming “Delete private tabs” to “Close private tabs” – with implied acknowledgement that users know: when private tabs are closed, your data is nuked.
2. Removing “Delete and open”, because users can go to PBM tab manager and tap the garbage bin icon. This two-step process is not as convenient as tapping on a notification, I agree. However, it doesn’t need to be convenient, because it’s not a primary action.
3. Removing “Open”, because Android system app switcher already handles opening the app. There’s no need to replicate it.

**TL;DR**
I would vote for only having one action: “Close private tabs”. When it’s pressed, your data is nuked. Want to reopen Firefox Preview after erasing? That’s what the home screen or app switcher is for.

**Comment by AmyYLee** [comment]
Hi, 

After discussion with @brampitoyo, here is the updated Notification menu. With one action only on tap: Close private tabs. 

Thanks!

![Screen Shot 2019-11-25 at 3 18 33 PM](https://user-images.githubusercontent.com/5992746/69574765-f0f01b80-0f96-11ea-9155-1501b2c0835d.png)

**Comment by NotWoods** [comment]
@AmyYLee I've found that the mask inside the circle icon is difficult to see as its displayed at a very small size in the notification bar. Can we change it to just the mask, as the gradient isn't preserved anyways?

**Comment by mcarare** [comment]
@AmyYLee & @brampitoyo Note that we cannot change the title displayed next to the icon, that text is the app name, and it's supplied by the system. 

![notif](https://user-images.githubusercontent.com/48995920/69636340-09c40400-105f-11ea-855c-749e05b11030.png)

**Comment by AmyYLee** [comment]
> @AmyYLee I've found that the mask inside the circle icon is difficult to see as its displayed at a very small size in the notification bar. Can we change it to just the mask, as the gradient isn't preserved anyways?

@NotWoods Yes that works for me. Please use the purple used for the circle for the mask colour (Dark theme should have a lighter purple that we are currently using as well).

**Comment by mcarare** [comment]
> @NotWoods Yes that works for me. Please use the purple used for the circle for the mask colour (Dark theme should have a lighter purple that we are currently using as well).

@AmyYLee This will also change the icon icon that appears in the system status bar.

![notif2](https://user-images.githubusercontent.com/48995920/69646514-9dea9700-1070-11ea-9d79-9cd82f64c800.png)

**Comment by AmyYLee** [comment]
> > @NotWoods Yes that works for me. Please use the purple used for the circle for the mask colour (Dark theme should have a lighter purple that we are currently using as well).
> 
> @AmyYLee This will also change the icon icon that appears in the system status bar.
> 
> ![notif2](https://user-images.githubusercontent.com/48995920/69646514-9dea9700-1070-11ea-9d79-9cd82f64c800.png)

Works for me. Thanks

**Comment by mcarare** [comment]
Thanks for the quick feedback @AmyYLee ! 😄

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest Nightly build 191202 with Google Pixel 2 (Android 9), Google Pixel 3 (Android Q), and Nexus 5 (Android 6.0.1).
The private notification has only the option to "Close private tabs". When tapping on it, the private opened tabs are closed, and the app isn't opened.
